File: app/src/main/java/io/github/skywalkerdarren/simpleaccounting/model/datasource/BillDataSource.java
package io.github.skywalkerdarren.simpleaccounting.model.datasource;

import java.util.List;
import java.util.UUID;

import io.github.skywalkerdarren.simpleaccounting.model.entity.Bill;
import io.github.skywalkerdarren.simpleaccounting.model.entity.BillInfo;

public interface BillDataSource {

    void getBillsCount(LoadBillCountCallBack callBack);

    void getBillsCount(int year, int month, LoadBillCountCallBack callBack);

    void getBillInfoList(int year, int month, LoadBillsInfoCallBack callBack);

    void getBill(UUID id, LoadBillCallBack callBack);

    void getsBills(int year, int month, LoadBillsCallBack callBack);

    void addBill(Bill bill);

    void delBill(UUID id);

    void updateBill(Bill bill);

    void clearBill();

    interface LoadBillCountCallBack {
        void onBillCountLoaded(Integer count);
    }

    interface LoadBillCallBack {
        void onBillLoaded(Bill bill);
    }

    interface LoadBillsCallBack {
        void onBillsLoaded(List<Bill> bills);
    }

    interface LoadBillsInfoCallBack {
        void onBillsInfoLoaded(List<BillInfo> billsInfo);
    }
}

File: app/src/main/java/io/github/skywalkerdarren/simpleaccounting/model/datasource/BillDataSource.kt
package io.github.skywalkerdarren.simpleaccounting.model.datasource

import io.github.skywalkerdarren.simpleaccounting.model.entity.Bill
import io.github.skywalkerdarren.simpleaccounting.model.entity.BillInfo
import java.util.*

interface BillDataSource {
    fun getBillsCount(callBack: (count: Int) -> Unit)
    fun getBillsCount(year: Int, month: Int, callBack: (count: Int) -> Unit)
    fun getBillInfoList(year: Int, month: Int, callBack: (billsInfo: List<BillInfo>?) -> Unit)
    fun getBill(id: UUID, callBack: (bill: Bill?) -> Unit)
    fun getsBills(year: Int, month: Int, callBack: (bills: List<Bill>?) -> Unit)
    fun addBill(bill: Bill)
    fun delBill(id: UUID)
    fun updateBill(bill: Bill)
    fun clearBill()
}